gnomad_qc/v3/sample_qc/sample_qc.py

```diff
@@ -462,24 +983,37 @@ def main(args):
         samples_to_drop = compute_related_samples_to_drop(
             relatedness.ht(),
             rank_ht,
-            args.kin_threshold,
+            args.second_degree_kin_cutoff,
             filtered_samples=filtered_samples
         )
+        samples_to_drop = samples_to_drop.key_by(s=samples_to_drop.s.s)
         samples_to_drop.checkpoint(pca_related_samples_to_drop.path, overwrite=args.overwrite, _read_if_exists=not args.overwrite)
         pop_pca_eigenvalues, pop_pca_scores_ht, pop_pca_loadings_ht = run_pca(args.include_unreleasable_samples, args.n_pcs, samples_to_drop)
         pop_pca_scores_ht.write(ancestry_pca_scores(args.include_unreleasable_samples).path, overwrite=args.overwrite)
         pop_pca_loadings_ht.write(ancestry_pca_loadings(args.include_unreleasable_samples).path, overwrite=args.overwrite)
-        with hl.utils.hadoop_open(ancestry_pca_eigenvalues_path(args.include_unreleasable_samples), mode='w') as f:
-            f.write(",".join([str(x) for x in pop_pca_eigenvalues]))
+
+        pop_pca_eigenvalues_ht = hl.Table.parallelize(
+            hl.literal(
+                [{'PC': i + 1, 'eigenvalue': x} 
+                 for i, x in enumerate(pop_pca_eigenvalues)
+                 ],
+                'array<struct{PC: int, eigenvalue: float}>'
+            )
+        )
+        pop_pca_eigenvalues_ht.write(
+            ancestry_pca_eigenvalues(args.include_unreleasable_samples).path,
+            overwrite=args.overwrite
+        )
 
     if args.assign_pops:
-        pop_ht, pops_rf_model = assign_pops(args.min_pop_prob, args.include_unreleasable_samples)
+        n_pcs = args.pop_n_pcs
+        pop_ht, pops_rf_model = assign_pops(args.min_pop_prob, args.include_unreleasable_samples, n_pcs=n_pcs, withhold_prop=args.withhold_prop)
         pop_ht = pop_ht.checkpoint(pop.path, overwrite=args.overwrite, _read_if_exists=not args.overwrite)
         pop_ht.transmute(
-            **{f'PC{i + 1}': pop_ht.pca_scores[i] for i in range(0, 10)}
-        ).export(pop_tsv_path)
+            **{f'PC{i + 1}': pop_ht.pca_scores[i] for i in range(0, n_pcs)}
```

Comment:
I think this can just be range(n_pcs) but definitely not a strong opinion on this one.